easybuild/easyblocks/t/tensorflow.py

```diff
@@ -803,13 +803,25 @@ def build_step(self):
                 self.target_opts.append('--config=cuda')
 
         # if mkl-dnn is listed as a dependency it is used. Otherwise downloaded if with_mkl_dnn is true
+        # Since 2.4.0 the mkl-config flag is not required as oneDNN is automatically used for x86 systems
+        # and MKL is no longer a dependency
         mkl_root = get_software_root('mkl-dnn')
         if mkl_root:
             self.target_opts.append('--config=mkl')
             env.setvar('TF_MKL_ROOT', mkl_root)
-        elif self.cfg['with_mkl_dnn']:
-            # this makes TensorFlow use mkl-dnn (cfr. https://github.com/01org/mkl-dnn)
-            self.target_opts.append('--config=mkl')
+        elif LooseVersion(self.version) < LooseVersion('2.4.0'):
+            # auto-enable use of MKL-DNN when possible if with_mkl_dnn is left unspecified
```

Comment:
@Flamefire Please clarify the comment to make it clear that this only applies to older TensorFlow versions, and also explain *why* it doesn't make sense to use `--config=mkl` for TensorFlow 2.4+ (including a ref to issue 2577)

Reply:
Isn't the one above enough?
> Since 2.4.0 the mkl-config flag is not required as oneDNN is automatically used for x86 systems and MKL is no longer a dependency

Reply:
Well, not really. It says the flag is no longer needed, but it actually way beyond that: not using that flag results in different installation w.r.t. threading model.
So there's a good reason why we don't use `--config=mkl` anymore: if we do, we end up with an installation that uses OpenMP threads, and then we hit the problem discussed in #2577 .
So this deserves a bit of additional info, and a pointer to https://github.com/easybuilders/easybuild-easyblocks/issues/2577

Maybe we should even go as far as also mentioning `--config=mkl_threadpool`, and why we don't use that either...
It's clear that what those configs imply is far from clear, otherwise we wouldn't be in this mess in the first place.

Also, the comment is a bit confusing: it mentions `mkl-config`, and it's not immediate clear that this relates to `--config=mkl`...